### PR TITLE
elan-init: remove dependency on x86, take 2

### PR DESCRIPTION
**Pull request by rami3l**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

cc https://github.com/Homebrew/homebrew-core/issues/127583

`elan-init` on ARM Macs is supported since 1.4.1 (see https://github.com/leanprover/elan/releases).

However, I'm forced to use LEAN4 ~~nightly~~ for the test since the old version (LEAN3) isn't natively supported on ARM Macs.

**Concern:** Should the `no ARM bottle` tag be removed?

### CODE REVIEW

**Comment by rami3l** [comment]
@SMillerDev That's true. However since `elan-init` is just a LEAN installer and now

- `elan` also provides prebuilt binaries for ARM Macs;
- `elan` can actually install a newer version of LEAN on ARM Macs (instead of being completely useless like before);

I think it's reasonable to remove this restriction, what do you think?

Many thanks!

File: Formula/elan-init.rb [hunk 1]
```diff
@@ -37,20 +35,22 @@ def install
   end
 
   test do
-    system bin/"elan-init", "-y"
+    ENV["ELAN_HOME"] = testpath/".elan"
+
+    system bin/"elan-init", "-y", "--default-toolchain=leanprover/lean4:nightly"
```
Comment (carlocab) [inline Formula/elan-init.rb:40]: Do we need to use the nightly toolchain for testing here?

File: Formula/elan-init.rb [hunk 2]
```diff
@@ -37,20 +35,22 @@ def install
...
-          intro h, cases h,
-          split, repeat { assumption }
-      end
+      example (a b : Prop) : a ∧ b -> b ∧ a := by
+          intro h; cases h with
+          | intro ha hb => constructor; exact hb; exact ha
     EOS
-    system bin/"lean", testpath/"hello.lean"
-    system bin/"leanpkg", "help"
+    system testpath/".elan/bin/lean", testpath/"hello.lean"
```
Comment (carlocab) [inline Formula/elan-init.rb:51]:
```suggestion
    system bin/"lean", testpath/"hello.lean"
```
We want to make sure the symlinks in `bin` still work.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/elan-init.rb [hunk 3]
```diff
@@ -37,20 +35,22 @@ def install
...
+
+    # Nightly versions of LEAN4 don't necessarily provide `leanpkg`.
+    # system testpath/".elan/bin/leanpkg", "help"
```
Comment (carlocab) [inline Formula/elan-init.rb:54]: Not keen on removing this check.

File: Formula/elan-init.rb (hunk 1 above)
Comment (rami3l) [reply Formula/elan-init.rb:40]: `--default-toolchain=leanprover/lean4:stable` will also do. The latest stable is a bit outdated but indeed it shouldn't matter that much for the purpose of this test.

File: Formula/elan-init.rb (hunk 2 above)
Comment (rami3l) [reply Formula/elan-init.rb:51]: Agreed. Will fix.

File: Formula/elan-init.rb (hunk 3 above)
Comment (rami3l) [reply Formula/elan-init.rb:54]: After some investigation I found out that `leanpkg` will not ship with LEAN4 in favor of `lake`. I'll replace `leanpkg` with `lake` instead.

**Review by carlocab — APPROVED** [review]
LGTM. Thanks!

**Comment by rami3l** [comment]
@carlocab This PR is ready to be merged. Thanks a lot!

**Pull request merged**